aptos-move/framework/aptos-framework/sources/event.move

```diff
@@ -54,4 +54,23 @@ module aptos_framework::event {
     public fun destroy_handle<T: drop + store>(handle: EventHandle<T>) {
         EventHandle<T> { counter: _, guid: _ } = handle;
     }
+
+    #[test_only]
+    use std::option::{Self, Option};
+
+    #[test_only]
+    public fun emitted_event<T: drop + store>(handle: &EventHandle<T>, sequence_number: u64): Option<T> {
+        let id = bcs::to_bytes(&handle.guid);
+        if (emitted_internal<T>(id, sequence_number)) {
+            option::some(emitted_event_internal<T>(id, sequence_number))
+        } else {
+            option::none()
+        }
+    }
+
```

Comment:
Can this also include a function like `get_emitted_event_unchecked()` that assumes some, and either unwraps it or asserts it is some?

Reply:
Inclined to leave the current set and go from there?

Reply:
This doesn't fit in with the model we landed on.